sdk/trace/batch_span_processor.go

```diff
@@ -108,15 +108,16 @@ func NewBatchSpanProcessor(e export.SpanBatcher, opts ...BatchSpanProcessorOptio
 	bsp.stopWait.Add(1)
 	go func(ctx context.Context) {
 		defer ticker.Stop()
+		batch := make([]*export.SpanData, 0, bsp.o.MaxExportBatchSize)
 		for {
 			select {
 			case <-bsp.stopCh:
-				bsp.processQueue()
+				bsp.processQueue(&batch)
```

Comment:
If the span producers pushing to the queue do so faster than the this can drain, it will cause this to hang.

I'm guessing we the `enqueue` method need to check if this `stopCh` is closed and not send any more spans while this flushes what has already been pushed.

Reply:
This is why I proposed adding something like the following to the `<-c.Ticker` case:

```go
select {
	case <-bsp.stopCh:
		return
	default:
}
```

But maybe putting this in `enqueue` is a better idea. This would also fix an issue of `enqueue` panicking when we call `Shutdown` in one goroutine (so the processor will finally close the `bsp.queue` channel in the ticker loop) while another goroutine causes the span data to be enqueued. Writing to the closed channel will panic, but adding the check for closed `bsp.stopCh` would fix it by silently dropping the span data.

Reply:
Done. Now we drain the queue, but ignore spans that are received after it stopping.

Reply:
Sounds good 👍